PhenomRaidTools/Source/Utils.lua

```diff
@@ -307,28 +307,57 @@ function PRT.ExchangeSpellIcons(s)
       return "|T"..(texture or "Interface\\Icons\\INV_MISC_QUESTIONMARK")..":16:16:0:0:64:64:6:58:6:58|t"
     end)
 end
-
-function PRT.AddCustomPlaceholdersToPlayerNames(token, t, customPlaceholders)
-  if customPlaceholders then
-    for _, customPlaceholder in ipairs(customPlaceholders) do
-      if customPlaceholder.name == token then
-        if customPlaceholder.type == "group" then
-          for i = #customPlaceholder.names, 1, -1 do
-            tinsert(t, strtrim(customPlaceholder.names[i], " "))
-          end
-        else
-          for _, name in ipairs(customPlaceholder.names) do
-            if (PRT.UnitInParty(name) or UnitExists(name)) and not UnitIsDead(name) then
+do
+  local placeholders = {global = {}, encounterSpecific = {}}
+  function PRT.SetupCustomPlaceholders()
+    placeholders = {global = {}, encounterSpecific = {}}
+    for id, customPlaceholder in ipairs(PRT.db.profile.customPlaceholders) do
+      -- use tables because you can have placeholders with same name (intended?)
+      if not placeholders.global[customPlaceholder.name] then
+        placeholders.global[customPlaceholder.name] = {id}
+      else
+        tinsert(placeholders.global[customPlaceholder.name], id)
+      end
+    end
+    for id, customPlaceholder in ipairs(PRT.currentEncounter.encounter.CustomPlaceholders) do
+      -- use tables because you can have placeholders with same name (intended?)
+      if not placeholders.encounterSpecific[customPlaceholder.name] then
+        placeholders.encounterSpecific[customPlaceholder.name] = {id}
+      else
+        tinsert(placeholders.encounterSpecific[customPlaceholder.name], id)
+      end
+    end
+  end
+  function PRT.AddCustomPlaceholdersToPlayerNames(token, t, global)
+    if not ((global and placeholders.global[token]) or (not global and placeholders.encounterSpecific[token])) then return end
+    for _,id in pairs(global and placeholders.global[token] or placeholders.encounterSpecific[token]) do
+      local temp = global and PRT.db.profile.customPlaceholders[id] or PRT.currentEncounter.encounter.CustomPlaceholders[id]
+      if not temp then return end -- should be useless nil check
+      if temp.type == "group" then
+        for i = #temp.names, 1, -1 do
+          tinsert(t, strtrim(temp.names[i], " "))
+        end
+      else
+        local found = false
+        local lastInPartyButDead
+        for _, name in ipairs(temp.names) do
+          if (PRT.UnitInParty(name) or UnitExists(name)) and UnitIsConnected(name) then
+            if not UnitIsDeadOrGhost(name) then
               tinsert(t, strtrim(name, " "))
+              found = true
               break
+            else -- so you can call for backup if needed
+              lastInPartyButDead = name
```

Comment:
The name should be trimmed here aswell i think. And with that we can do it directly when we get into this branch instead of writing it two times.

Reply:
trimming it later saves resources, because the name might not be used at all

Reply:
noticed that the previous line didn't actually use lastInPartyButDead var when it should have, fixed in https://github.com/PhenomDevel/PhenomRaidTools/pull/32/commits/be232421ea3b3565761dbe90c0c3e2afd2e690a5